src/dnsmasq_interface.c

```diff
@@ -616,8 +635,8 @@ static void detect_blocked_IP(const unsigned short flags, const char* answer, co
 
 		// Update status
 		query_externally_blocked(queryID, QUERY_EXTERNAL_BLOCKED_IP);
+		return;
```

Comment:
Why are these returns added? They are unnecessary as the function exits here anyways.